Adding lncli access to Settings tab

diofeher opened the pull request:
Related to #54 

This pull request is to see if I'm heading to the right direction. The interface should be used to interact **only** with lncli?

What about creating a separate tab on the left menu to interact with the CLI (separating from settings, doesn't make sense to put it there, or at least rename Settings to "Advanced Use").

casesandberg committed:
Fetch Witness Address

casesandberg committed:
Generate Payment Request

casesandberg committed:
Add Schema and Single Type to `redux-grpc-middleware`

casesandberg committed:
Hook Transactions Up

casesandberg committed:
Hook Up Sidebar

casesandberg committed:
Rearrange Build Scripts

casesandberg committed:
Initialize `lightning-currency`

casesandberg committed:
Add Back in React to Root

casesandberg committed:
Change to Hosted Eslint

casesandberg committed:
Remove Nav?

casesandberg committed:
Update Install Instructions

casesandberg committed:
Add Common Errors to Readme

casesandberg committed:
GRPC Fix

casesandberg committed:
Add Empty Transactions

casesandberg committed:
Add Copied to Clipboard Notification

casesandberg committed:
Add Notification When Copying Bitcoin Address

casesandberg committed:
Remove Currency Selector For Now

casesandberg committed:
Have Core Export Reducer and Selectors

casesandberg committed:
Init Pay Reducer

casesandberg committed:
Check Payment Request Scaffolding

casesandberg committed:
Wire Up Channels

casesandberg committed:
Empty Channels

casesandberg committed:
Update Copy

casesandberg committed:
Create Channel Link

casesandberg committed:
Add Create Channel Page

casesandberg committed:
Create Channel Actions

casesandberg committed:
New New Autocomplete Bug?

casesandberg committed:
Update LND!

casesandberg committed:
Make Payment

casesandberg committed:
Hook Up Wallet

casesandberg committed:
Fetch Balances on Page Load

casesandberg committed:
Add Back Transaction Status #whoops

casesandberg committed:
Init Settings Page

casesandberg committed:
Put Back Settings Page

casesandberg committed:
Add QR Popup

casesandberg committed:
Remove Legacy Files

casesandberg committed:
Eslint Fixes

casesandberg committed:
Check Account every 20 seconds

casesandberg committed:
Add Back Transactions and Invoices Streams

casesandberg committed:
Close Channel

casesandberg committed:
Subscribe Payments

casesandberg committed:
Cleanup Old Files

casesandberg committed:
Remove Primitive TrafficLights

casesandberg committed:
<App /> to stateless

casesandberg committed:
Update Router Deps

casesandberg committed:
Update LND Binary

casesandberg committed:
Transactions and Wallet Is In Satoshi Now

casesandberg committed:
Add Syncing Animation

casesandberg committed:
Add Traffic Light Padding

casesandberg committed:
Add Back Satoshi Conversion for Account

casesandberg committed:
Pointer Nav Footer Account to Accounts Page

casesandberg committed:
Update Drag Settings

casesandberg committed:
Disable Selection of Non-Essential Items

casesandberg committed:
Fix Package Script

casesandberg committed:
Comment Out `sendPayment`

casesandberg committed:
Streams Notifications Update

casesandberg committed:
Streams for Send Payment

casesandberg committed:
Close Channel Stream Call

casesandberg committed:
Transactions Errors

casesandberg committed:
Add `txindex` to btcd

casesandberg committed:
Create Channel Stream

casesandberg committed:
Close Channel

casesandberg committed:
Make `listPayments` call

casesandberg committed:
Fetch Channels on Poll

casesandberg committed:
Fetch Channels on Lightning Payment Recieved

casesandberg committed:
Fix Accounts Header Flex Issue

casesandberg committed:
Fetch Pending Channels

casesandberg committed:
Add Active to Channels

casesandberg committed:
Close Channel Icon on Hover

casesandberg committed:
Errors and Success on Opening Channel

casesandberg committed:
Pending Channel Greyed Out

casesandberg committed:
Change Opening Channel Message

casesandberg committed:
Infinte Scrolling Logs

casesandberg committed:
Kill Running Processes On App Quit

casesandberg committed:
Fill Out Amount on PaymentRequest when Making Payment

casesandberg committed:
`Catch` a Bunch of Errors

casesandberg committed:
Remove Subscribe Payments

casesandberg committed:
Fetch Account on New Transactions & Invoices

casesandberg committed:
Remove Unnecessary Catches

casesandberg committed:
Release All Callbacks on App Reload

casesandberg committed:
Catch Transaction Errors

casesandberg committed:
Dedupe by Channel Point

casesandberg committed:
Active Label

casesandberg committed:
Add `minimizeOnClose`

casesandberg committed:
Don't Kill Child Processes

casesandberg committed:
Change CreateChannel Label

casesandberg committed:
Fetch Channel on Invoice Complete

casesandberg committed:
Add Pending Channel ID As Pending

casesandberg committed:
Make InfiniteScroll to Handle Height

casesandberg committed:
Catch If Window Doesn't Exist

casesandberg committed:
Catch GRPC Calls

casesandberg committed:
Fix Spawn Process Path

casesandberg committed:
Fix Error Messages

casesandberg committed:
Stop Devtools on Load in Dev

casesandberg committed:
Build Working

casesandberg committed:
Fix CWD Paths For Packaged App

casesandberg committed:
Package Electron All from Root

casesandberg committed:
Package All!

casesandberg committed:
Fix Infinite Scroll Height

casesandberg committed:
Finalize Packaging for Prod

casesandberg committed:
Remove Minding Address in Prod

casesandberg committed:
Add Remote Debugging for Electron in Prod

casesandberg committed:
Make Vendor File in Prod

casesandberg committed:
Make LND Call in Electron not Webpack

casesandberg committed:
Fill UI on Failures

casesandberg committed:
Add Already Running Log

casesandberg committed:
Remove Process Details from Log

casesandberg committed:
Add Deadline to Calls

casesandberg committed:
Change Deadline to 30 seconds

casesandberg committed:
Kill Processes on App Quit

casesandberg committed:
Call Reducer Error on Catch

casesandberg committed:
Add Loading Icons

casesandberg committed:
Get Icon to Work?

casesandberg committed:
Add 9997 to Readme

casesandberg committed:
Add Icons

casesandberg committed:
Update `rpc.proto`

casesandberg committed:
Update Binaries and Move into Darwin

casesandberg committed:
Add Platform Path for Binaries

casesandberg committed:
Delete then Build

casesandberg committed:
Simplify Start Pack Command

casesandberg committed:
Add Platform to Bin Path

casesandberg committed:
Add Entire Error Message on Start Pack

casesandberg committed:
Catch Main Process Errors?

casesandberg committed:
Add New Binaries + Windows EXE

casesandberg committed:
Add Asset

casesandberg committed:
Change Logging

casesandberg committed:
Add Prod Menu

casesandberg committed:
Add Select All

casesandberg committed:
Change Debug Level

casesandberg committed:
Fix Loading Icon Import

casesandberg committed:
Try Catch on `execFile` processes

casesandberg committed:
Get Recent 500 logs

casesandberg committed:
String Errors

casesandberg committed:
Only Apply Menu on Prod

casesandberg committed:
Check If Server is Running Before Making GRPC Calls

casesandberg committed:
Start Streams when Server is Running

casesandberg committed:
Reorder Sent Flags Assignment to Stop Duplicates

casesandberg committed:
Show Syncing when Server is Loading Up

casesandberg committed:
Batch Running Process Logs

casesandberg committed:
Send Request Action before Rejecting If not Ready

casesandberg committed:
Fetch Channels on Payment Sent

casesandberg committed:
Check for Peer to Determine Force on Close Channel

casesandberg committed:
Add Push to Router

casesandberg committed:
Revert Close Channel to Just add Force True

casesandberg committed:
Add CID Label

casesandberg committed:
Fix Pending Channels Populate

casesandberg committed:
Only Force if No Peer

casesandberg committed:
Fetch Account Immediately After Server Starts

casesandberg committed:
Fetch Account on Payment Success

casesandberg committed:
Don't Show Close on Pending Channels

casesandberg committed:
Remove Extra Catch Stopping On-Chain Payments

casesandberg committed:
Only Add CID Labels to channels with ids

casesandberg committed:
Spelling Fixes

casesandberg committed:
Keep Newer Channel Information

casesandberg committed:
Remove Extra Balance Zero

casesandberg committed:
Resolve `createChannel`

casesandberg committed:
Put Channel in Closing Status

casesandberg committed:
Update RPC and Binaries

casesandberg committed:
Update Binary Flags

casesandberg committed:
Update Channel Status Styling

casesandberg committed:
Fetch Account and Channels on Server Startup

casesandberg committed:
Add Limbo Balance to Wallet

casesandberg committed:
On Send Logs if There are Logs

casesandberg committed:
Separate Out Pending Channels for Clashing

casesandberg committed:
Add `parseFloat` to fix floating-point multiplication bug

casesandberg committed:
Updated Github Logo

casesandberg committed:
Add Linux Distro

casesandberg committed:
Merge pull request #2 from lightninglabs/update-rpc

Update RPC

casesandberg committed:
Merge branch 'master' into fix-floating-point

casesandberg committed:
Update Channels on Payment

casesandberg committed:
Add Production Logs

casesandberg committed:
Merge pull request #3 from lightninglabs/fix-floating-point

Laolu Fixes

casesandberg committed:
Move Desktop to Apps

casesandberg committed:
Fix Desktop Pathing

casesandberg committed:
Change Startup Message

casesandberg committed:
Update Bin Path

casesandberg committed:
Cleanup Args

casesandberg committed:
Merge pull request #4 from lightninglabs/apps-dir

Apps Dir

bryanvu committed:
Updated to lnd commit b1205e71ecb46, updated protos and RPC calls accordingly.

bryanvu committed:
Updated to lnd alpha 0.3, added TLS support

Roasbeef committed:
pay: refresh the account page, and bottom toolbar on payment success

Roasbeef committed:
pay: correct check for payment error on payment attempt

Previously, we checked against a null value, while within the proto
itself, the error is actually a string. To fix this, we properly check
against an empty string. With this commit, payment errors will now
properly display.

Roasbeef committed:
pay: pass in a proper error to rejectError

In this commit, we ensure that upon a payment error, is properly
displayed within the UI itself.

Roasbeef committed:
pay: properly execute the resolveSuccess callback on payment success

With this commit, we ensure that the trigger to update the UI upon
payment success is actually triggered. Previously, we would simply
reference the function rather than calling it directly.

Roasbeef committed:
pay: slightly modify copy for diff between on-chain vs off-chain

Roasbeef committed:
common: change accounts on toolbar to channels

Roasbeef committed:
accounts: modify copy for description of channels

bryanvu committed:
Cross-platform bugfixes

 - Switched to platform-native title bar, removed electron "traffic lights".
 - Removed platform default drop-down menus.
 - Added title bar icon.
 - Removed draggability from left sidebar.
 - Fixed Windows path to lnd binary.
 - Enabled autopilot.

bryanvu committed:
lnd: updated to commit 6253419, connect to lightning.computer

bryanvu committed:
logs: switched from childprocess.execFile() to cp.spawn()

With execFile, the process was shutting down after the logs reached a
certain size due to apparent 200kb limit or after a particularly large
batch of logs were written. With spawn, the logs are unlimited.

bryanvu committed:
icons: switched to official lightning labs logo

Switched logos for task bar icons, favicon and splash screen (svg
formatted).

bryanvu committed:
lnd: updated to commit f20cb89

Roasbeef committed:
README: correct link to github repo

Roasbeef committed:
README: update with description

Roasbeef committed:
add GPLv3 license

willpiers committed:
README: removed step to run npm install

hokkaidosuneku committed:
Allowed displaying complete TxID, note with word-wrap and PubKey in wallet page with word-wrap

Fixed spacing and formatting

Updated with cleaner code for conditional rendering of transaction note

casesandberg committed:
Fetch Balance when Fetching Accounts

bryanvu committed:
lnd: update to lnd commit d89491

hokkaidosuneku committed:
Added commas to amounts on Wallet, transactionslist and channellist along with currency unit

Fixed spacing for onFetchAccount

Roasbeef committed:
bump version to 0.1.1-alpha

bryanvu committed:
README: updated readme to remove lnd build and to add package-electron

Commit:
Fix spacing

There are a variety of spacing issues throughout the code. Most of
these issues deal with the first line in the file being a newline.

This commit removes unnecessary spaces and adds them when absolutely
necessary.

Commit:
Remove unnecessary commented-out code

Commit:
Fix ESLint rules

There are a variety of ESLint rules broken in the codebase. The
majority of issues include:

- Tab spacing
- Missing a trailing comma

Other issues fixed included removing unnecessary commented-out code,
consolidating ESLint rule exceptions, and cleaning up code.

bryanvu committed:
doc: created issue template

As suggested by @thejourneydude

hokkaidosuneku committed:
Added right-click context menu with copy and select all for payrequest, bitcoin wallet address and settings page

Fixed commented code

hokkaidosuneku committed:
Add paste option for pasting wallet address and amount in Pay screen

hokkaidosuneku committed:
Added paste context menu for create channel form

Fixed spacing

hokkaidosuneku committed:
Prevent wrapping of currency unit to the next line and fix flex on Wallet and Transaction list item amounts.

Commit:
Fix race condition between start-electron and start-webpack scripts

There is a race condition between start-electron and start-webpack.
When we run npm start, both of these scripts run concurrently. The
start-electron script relies on the start-webpack script to have
completed, thus if the start-electron script is finished BEFORE the
start-webpack script, then the app will open with a blank screen.

To solve this, I added a sleep command to the start-electron script.
This should ensure that the start-webpack script is complete and that
the start-electron can work properly.

For more information on this bug, see line 143 in
apps/desktop/main.dev.js. This line sends a HTTP request to
localhost:4152, our Node server. If the Node server hasn’t started, the
app will open up with a blank page.

Commit:
Remove mention of bug fix

This bug should now be fixed and no longer requires a comment in the
readme.

bryanvu committed:
Revert "Remove mention of bug fix"

This reverts commit ac5d5ace932f3e9b9204472f4e530a9cee1bbf77.

bryanvu committed:
Revert "Fix race condition between start-electron and start-webpack scripts"

This reverts commit 7e6cd14f82a9dd65c1cf19a7e4198b06572cd811.

bryanvu committed:
auth: add support for macaroons

Previously, the app was running lnd with the —no-macaroons option. This
commit adds the ability for the app to send macaroons to lnd with each
gRPC call.

bryanvu committed:
auth: add support for macaroons

Previously, the app was running lnd with the —no-macaroons option. This
commit adds the ability for the app to send macaroons to lnd with each
gRPC call.

bryanvu committed:
dev: remove configuration for dev mode, use local lnd.conf

Previously, when running the app in dev mode using “npm start”, the app
would attempt to run on simnet, but would be unable to do so due to the
lack of a packaged btcd. This commit removes the default neutrino
configuration and will use the configuration in the local lnd.conf.
This should give developers greater flexibility when building and
testing app features.

bryanvu committed:
readme: removed section headers in lnd.conf

bryanvu committed:
Disable node integration

As suggested by @kewde in issue #75, disabled nodeIntegration.

bryanvu committed:
Updated to lnd commit 0ec22cc

bryanvu committed:
Added --nowalletencryption flag

Roasbeef committed:
update version to 0.1.2

bryanvu committed:
Revert "auth: add support for macaroons"

This reverts commit 4663b07438b5931d7756bbbaa71d2ad44b4029e1.

bryanvu committed:
lnd: update lnd to commit 5dbf508

hokkaidosuneku committed:
Fixed log height and width adjustment on resize of screen in settings page

bryanvu committed:
lnd: update to commit c0b8c29

Roasbeef committed:
update version to 0.1.3

bryanvu committed:
lnd: update to commit 4d6cd2e

Roasbeef committed:
bump version to 0.1.3.1

Roasbeef committed:
package: modify latest version to comply with npm's semantic version

The default semantic version parsing on npm, seems to apply a regex
similar to: `\d+\.\d+\.\d+`. As a result, our latest release (a small
compat patch on top of the prior one) fails to build on the command
line. With this change, we restore the ability to build.

diofeher committed:
Adding subdependencies and Terminal component

diofeher committed:
Adding Terminal component

diofeher commented:
<img width="1074" alt="screen shot 2017-12-08 at 10 16 57" src="https://user-images.githubusercontent.com/161360/33767883-8644c28c-dc02-11e7-8592-b542227ee2c7.png">

diofeher commented:
If **lncli** is the only command necessary to be run here, we can remove the dependency of react-bash.

bryanvu commented:
Thanks @diofeher! I'm going to do a bit more testing but we only need lncli, so I would recommend removing the option for any other commands.

bryanvu commented:
Oh, and about your other suggestion, I think that renaming settings to "Advanced" makes sense. I think that tab could contain the Logs and the Console. We can take out the "Settings" for now and add it back in later.

yate committed:
add qrcode to payment request

yate committed:
revert package.json files

yate committed:
remove console.log

yate committed:
replace lightning:// with lightning:

yate committed:
make qr images sharper

diofeher committed:
Removing react-bash and reworking the UI

diofeher commented:
Thanks for the comments @bryanvu !

Here's the updated UI:

<img width="1392" alt="screen shot 2017-12-11 at 16 52 38" src="https://user-images.githubusercontent.com/161360/33850872-0aba80c2-de94-11e7-83fc-9ba15db78a25.png">

diofeher committed:
Getting feedback after sending the lncli command

diofeher commented:
I've added the ability to see the command you sent on the logs and I've added a timestamp.

<img width="784" alt="screen shot 2017-12-12 at 11 37 36" src="https://user-images.githubusercontent.com/161360/33890138-19ca8a38-df31-11e7-9cd0-c22ecc83715c.png">

diofeher committed:
Adding the ability to pass arguments in npm start to lnd command

diofeher committed:
Updating README

diofeher committed:
Removing unnecessary code

diofeher committed:
Removing console.log

bulenttastan committed:
Disable amount field for invoice payment requests

bryanvu committed:
payreq: replace 'lightning://' with 'lightning' on payment requests'

diofeher committed:
Master merge

diofeher committed:
Reverting package.json

diofeher committed:
More cleanup

diofeher commented:
@bryanvu anything missing to merge this?

bryanvu commented:
@diofeher, huge apologies for the slow response on this. Am just recovering from the holidays and work on a couple other projects. I just pulled this and built it (npm run package-electron), and when I enter anything in the "lncli" text box, I'm not getting any additional output in the console. I tried "getinfo" and "help". However, when I run in dev mode (npm start), it does work. Any thoughts on why it doesn't work with the packaged version?

diofeher committed:
Packaging lncli with electron by default

diofeher committed:
Bundling lncli together with ln app

diofeher commented:
@bryanvu I need one favor, to add lncli binaries to windows and linux :)

diofeher commented:
@tanx any reason to close it?

tanx commented:
@diofeher sorry I don’t even remember closing this.
